### PR TITLE
Add the channelID property to bot connector calls.

### PR DESCRIPTION
**Pull request by Jeffders**
This change adds the channelID property to the calls on the OAuthClient on the connector library.

### CODE REVIEW

File: CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthClient.cs [hunk 1]
```diff
@@ -37,6 +37,11 @@ public partial class OAuthClient : ServiceClient<OAuthClient>, IOAuthClient
         /// </summary>
         public virtual IOAuthApi OAuthApi { get; private set; }
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthClient.cs]: Should we mark this obsolete?

File: CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthClientExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+namespace Microsoft.Bot.Connector
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthClientExtensions.cs]: copyright headers

File: CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthClient.cs (hunk 1 above)
Comment (carlosscastro) [reply CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthClient.cs]: My thinking is that if at some point we do a cleanup that include breaking changes, removing obsolete stuff should leave nice abstractions. Seems like the rest of your PR is making the old IOauthApi obsolete, so if possible take it all the way, unless I'm missing something.

File: CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthApi.cs [hunk 3]
```diff
@@ -23,7 +23,7 @@ namespace Microsoft.Bot.Connector
     /// <summary>
     /// OAuthClient operations.
     /// </summary>
-    public partial class OAuthApi : IServiceOperations<OAuthClient>, IOAuthApi
+    public partial class OAuthApi : IServiceOperations<OAuthClient>, IOAuthApiEx
```
Comment (carlosscastro) [inline CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthApi.cs]: So the methods for the old IOAuthAPI implementation are there but we removed it from line 26, so the OAuthAPI class does not implement the interface anymore. could any backward compat issues stem from that? i.e. from user written code that used OAuthAPI and depended on the fact that it implemented IOAuthAPI interface? Even though it sounds unlikely, if it is a backward compat violation we must fix it. Seems like that case could be mitigated by simply leaving this as :

`public partial class OAuthApi : IServiceOperations<OAuthClient>, IOAuthApiEx, IOAuthApi`

Thoughts?

**Comment by carlosscastro** [comment]
Looks good overall, but needs the missing copyrights header and wanted to discuss a few minor details.

File: CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthClient.cs (hunk 1 above)
Comment (Jeffders) [reply CSharp/Library/Microsoft.Bot.Connector.Shared/OAuthAPI/OAuthClient.cs]: Can do.

---
In reply to: [330821759](https://github.com/microsoft/BotBuilder-V3/pull/165#discussion_r330821759) [](ancestors = 330821759)

**Comment by Jeffders** [comment]
Originally I had IOAuthApiEx derive from IOAuthApi because some of the methods did not need to be obsoleted, but I switched to just a complete new interface so that we could obsolete the entire other interface some day if we so desired.

---
In reply to: [537733625](https://github.com/microsoft/BotBuilder-V3/pull/165#issuecomment-537733625) [](ancestors = 537733625)

**Pull request merged**